src/OpenSolid/Sphere3d.elm

```diff
@@ -0,0 +1,500 @@
+
+    Sphere3d.mirrorAcross Plane3d.xy exampleSphere
+    --> Sphere3d.with
+    -->     { centerPoint =
+    -->         Point3d.fromCoordinates ( 1, 2, -1 )
+    -->     , radius = 3
+    -->     }
+
+-}
+mirrorAcross : Plane3d -> Sphere3d -> Sphere3d
+mirrorAcross plane =
+    let
+        mirrorPoint =
+            Point3d.mirrorAcross plane
+    in
+    \sphere ->
+        with
+            { centerPoint = mirrorPoint (centerPoint sphere)
+            , radius = radius sphere
+            }
+
+
+{-| Take a sphere defined in global coordinates, and return it expressed in
+local coordinates relative to a given reference frame.
+
+    localFrame : Frame3d
+    localFrame =
+        Frame3d.atPoint
+            (Point3d.fromCoordinates ( 1, 2, 3 ))
+
+    Sphere3d.relativeTo localFrame exampleSphere
+    --> Sphere3d.with
+    -->     { centerPoint =
+    -->         Point3d.fromCoordinates ( 0, 0, -2 )
+    -->     , radius = 3
+    -->     }
+
+-}
+relativeTo : Frame3d -> Sphere3d -> Sphere3d
+relativeTo frame sphere =
+    with
+        { centerPoint = Point3d.relativeTo frame (centerPoint sphere)
+        , radius = radius sphere
+        }
+
+
+{-| Take a sphere considered to be defined in local coordinates relative to a
+given reference frame, and return that sphere expressed in global coordinates.
+
+    localFrame : Frame3d
+    localFrame =
+        Frame3d.atPoint
+            (Point3d.fromCoordinates ( 1, 2, 3 ))
+
+    Sphere3d.placeIn localFrame exampleSphere
+    --> Sphere3d.with
+    -->     { centerPoint =
+    -->         Point3d.fromCoordinates ( 2, 4, 4 )
+    -->     , radius = 3
+    -->     }
+
+-}
+placeIn : Frame3d -> Sphere3d -> Sphere3d
+placeIn frame sphere =
+    with
+        { centerPoint = Point3d.placeIn frame (centerPoint sphere)
+        , radius = radius sphere
+        }
+
+
+{-| Get the minimal bounding box containing a given sphere.
+
+    Sphere3d.boundingBox exampleSphere
+    --> BoundingBox3d.with
+    -->     { minX = -2
+    -->     , maxX = 4
+    -->     , minY = -1
+    -->     , maxY = 5
+    -->     , minZ = -2
+    -->     , maxZ = 4
+    -->     }
+
+-}
+boundingBox : Sphere3d -> BoundingBox3d
+boundingBox sphere =
+    let
+        r =
+            radius sphere
+
+        ( cx, cy, cz ) =
+            Point3d.coordinates
+                (centerPoint sphere)
+    in
+    BoundingBox3d.with
+        { minX = cx - r
+        , maxX = cx + r
+        , minY = cy - r
+        , maxY = cy + r
+        , minZ = cz - r
+        , maxZ = cz + r
+        }
+
+
+{-| Check if a sphere contains a given point.
+
+    Sphere3d.contains
+        (Point3d.fromCoordinates (4, 2, 1))
+        exampleSphere
+    --> True
+
+    Sphere3d.contains
+        (Point3d.fromCoordinates (4.00001, 2, 1))
+        exampleSphere
+    --> False
+
+-}
+contains : Point3d -> Sphere3d -> Bool
+contains point sphere =
+    Point3d.squaredDistanceFrom point (centerPoint sphere) <= radius sphere ^ 2
```

Comment:
I'm a bit nervous about using `^` since it (unsurprisingly) looks like it compiles down to a `Math.pow` call, which seems generally slower than a simple multiplication, especially on older browsers (see https://stackoverflow.com/questions/26593302/whats-the-fastest-way-to-square-a-number-in-javascript). That's why I use a temporary `r` variable in similar places - perhaps a premature optimization, but I think the result is still quite readable...